### PR TITLE
Add security fixes

### LINKED ISSUE #15: Validate the form
Right now we are not [validating](https://developer.wordpress.org/plugins/security/data-validation/) the form in any capacity. At the very least we need to make the name of the theme required and [sanitize](https://developer.wordpress.org/plugins/security/securing-input/) all the inputs submitted by the user.

### PR DESCRIPTION
**Pull request by scruffian**
This adds some security fixes:

- Sanitize all inputs
- Add another user capability check
- Add a nonce check
- Add some error/success meessages

Fixes #15

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -315,7 +323,38 @@ function blockbase_create_theme_menu() {
...
+
+		return add_action( 'admin_notices', 'create_blockbase_child_admin_notice_success' );
 		gutenberg_edit_site_export_theme( $_GET['theme'] );
 	}
 }
 add_action( 'admin_init', 'blockbase_save_theme');
+
+function create_blockbase_child_admin_notice_error() {
+	$class = 'notice notice-error';
+	$message = __( 'Please specify a theme name.', 'create-blockbase-theme' );
```
Comment (scruffian) [inline index.php:349]: Sometimes you can get in here for other errors, but only ones that users shouldn't hit unless they are doing something dodgy!

**Comment by MaggieCabrera** [comment]
Should we add a `require` attr to the theme name input too?

**Comment by scruffian** [comment]
Done!

**Comment by MaggieCabrera** [comment]
Something is not working, I get the success message but I don't get a zip!

**Comment by scruffian** [comment]
> Something is not working, I get the success message but I don't get a zip!

are you running GB trunk?

**Comment by MaggieCabrera** [comment]
> are you running GB trunk?

yep!

**Comment by scruffian** [comment]
I dont think this will work against trunk because of the key change from user to custom - it's probably throwing a fatal error.

**Comment by MaggieCabrera** [comment]
I just tried with the latest version from the dotorg repo and it's the same. It's 12.0.0, on a fresh site too.

**Comment by scruffian** [comment]
i just pushed a fix. Such a silly mistake! 🤦

**Review by MaggieCabrera — APPROVED** [review]
LGTM now!

**Pull request merged**